src/client/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -23,7 +23,7 @@ namespace Microsoft.Identity.Client
 {
     internal sealed class ApplicationConfiguration : IAppConfig
     {
-        public ApplicationConfiguration(bool isConfidentialClient) 
+        public ApplicationConfiguration(bool isConfidentialClient, bool isManagedIdentity = false) 

```

Comment:
Pls use an enum instead of 2 flags. There are 12 references to this, it's not a big refactoring.

Reply:
The concern with 2 flags is that the state [ConfidentialClient = true, ManagedIdentity = true] is not possible, but MSAL developers might forget this.

Reply:
For example https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenSilentParameterBuilder.cs#L209 - this is a bug, as the error message is wrong.

Reply:
Updated. Makes sense.